haystack/schema.py

```diff
@@ -55,7 +55,8 @@ def __init__(
             score: Optional[float] = None,
             meta: Dict[str, Any] = None,
             embedding: Optional[np.ndarray] = None,
-            id_hash_keys: Optional[List[str]] = None
+            id_hash_keys: Optional[List[str]] = None,
+            id_hash_from:  Optional[List[Literal["content", "meta"]]] = None
```

Comment:
I'd say let's go for a (small) breaking change here and rather update `id_hash_keys` to the new functionality and not introduce a second parameter. If users really want to use custom strings for hashing, they can still do it outside of haystack and provide a custom id directly. 

To facilitate the migration of users, we can:
1. add a check in the init if `id_hash_keys is in ("content", "meta")`
2. if false: throw a clearly formulated exception a la (f"You passed custom strings {id_hash_keys} to `id_hash_keys` which is deprecated. Supply instead a list of Document's attribute names that the id should be based on (e.g. ["content", "meta"]). See github.com/pull/ xxx for details)